batchflow/tests/classification_metrics_test.py

```diff
@@ -0,0 +1,175 @@
+def metrics_name(request):
+    return request.param
+
+def test_diff_shapes_two_classes(metrics_name):
+    """Testing different shape"""
+    y_true_1, y_pred_1 = np.array([0, 1]), np.array(1)
+    y_true_2, y_pred_2 = np.array([[0, 1], [1, 0]]), np.array([0, 1])
+    y_true_3, y_pred_3 = np.array([[[0, 1], [1, 0]], [[1, 1], [0, 0]]]), np.array([[0, 1], [1, 0]])
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_1, y_pred_1, fmt='labels', num_classes=2), metrics_name)()
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_2, y_pred_2, fmt='labels', num_classes=2), metrics_name)()
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_3, y_pred_3, fmt='labels', num_classes=2), metrics_name)()
+
+def test_diff_shapes_multiclass(metrics_name):
+    """Testing different shapes for multiclass"""
+    y_true_1, y_pred_1 = np.array(2), np.array([[0.8, 0.1, 0.1], [0.1, 0.1, 0.8]])
+    y_true_2 = np.array([[[0, 1, 0], [0, 1, 0], [1, 0, 0], [0, 0, 1]], [[1, 0, 0], [1, 1, 0], [1, 0, 1], [0, 0, 1]]])
+    y_pred_2 = np.array([[0.1, 0.8, 0.1], [0.1, 0.8, 0.1], [0.8, 0.1, 0.1], [0.8, 0.1, 0.1]])
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_1, y_pred_1, fmt='proba', axis=1, num_classes=3), metrics_name)()
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_2, y_pred_2, fmt='proba', axis=1, num_classes=3), metrics_name)()
+
+@pytest.mark.parametrize('metrics, metrics_shape', [
+    ('accuracy', ()),
+    ('f1_score', (1, 1)),
+    ('true_positive_rate', (1, 1)),
+    ('false_positive_rate', (1, 1))
+])
+def test_single_value_two_class(metrics, metrics_shape):
+    """Test on various metrics: output single value in case of two class classification"""
+    y_true, y_pred = np.random.choice([0, 1], size=(5,)), np.random.choice([0, 1], size=(5,))
+    test_shape = getattr(cm(y_true, y_pred, fmt='labels', num_classes=2), metrics)().shape
+    assert test_shape == metrics_shape
+
+@pytest.mark.parametrize('metrics, metrics_shape', [
+    ('accuracy', ()),
+    ('f1_score', (1, 1)),
+    ('true_positive_rate', (1, 1)),
+    ('false_positive_rate', (1, 1))
+])
+def test_vector_multiclass(metrics, metrics_shape):
+    """Test on metrics: output single value in case of multiclass"""
+    y_true, y_pred = np.array([2, 1]), np.array([[0.1, 0.1, 0.8], [0.1, 0.8, 0.1]])
+    test_shape = getattr(cm(y_true, y_pred, fmt='proba', axis=1, num_classes=2), metrics)().shape
+    assert test_shape == metrics_shape
+
+@pytest.mark.parametrize('metrics, metrics_shape', [
+    ('accuracy', (3, )),
+    ('f1_score', (3, 1)),
+    ('true_positive_rate', (3, 1)),
+    ('false_positive_rate', (3, 1))
+])
+def test_vector_batches_two_class(metrics, metrics_shape):
+    """Test on metrics: output vector with batch shape if input is a multidimensional array, two class"""
+    y_true = np.array([[[1, 1], [0, 1]], [[0, 1], [1, 1]], [[1, 0], [1, 1]]])
+    y_pred = np.array([[[0, 1], [1, 1]], [[1, 0], [0, 0]], [[0, 0], [0, 1]]])
```

Comment:
Either don't use explicit values (and only specify shape for arrays), or put those values in separate fixture (and use it in as many test functions as possible)